code/modules/mob/dead/observer/observer.dm

```diff
@@ -297,30 +297,43 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
 /mob/dead/observer/proc/show_me_the_hud(hud_index)
 	var/datum/atom_hud/H = huds[hud_index]
 	H.add_hud_to(src)
-	data_hud_seen = hud_index
+
+/mob/dead/observer/proc/remove_the_hud(hud_index) //remove old huds
+	var/datum/atom_hud/H = huds[hud_index]
+	H.remove_hud_from(src)
 
 /mob/dead/observer/verb/toggle_medHUD()
 	set category = "Ghost"
-	set name = "Toggle Medic/Sec/DiagHUD"
-	set desc = "Toggles the medical HUD."
+	set name = "Toggle Medic/Sec/Diag/All HUDs"
+	set desc = "Toggles the HUDs."
 	if(!client)
 		return
-	if(data_hud_seen) //remove old huds
-		var/datum/atom_hud/H = huds[data_hud_seen]
-		H.remove_hud_from(src)
 
 	switch(data_hud_seen) //give new huds
 		if(0)
+			data_hud_seen = 1
 			show_me_the_hud(DATA_HUD_SECURITY_ADVANCED)
 			to_chat(src, "<span class='notice'>Security HUD set.</span>")
-		if(DATA_HUD_SECURITY_ADVANCED)
+		if(1)

```

Comment:
Shouldn't you be using the HUD defines here instead of magical numbers?

Reply:
I could, but it would make no difference. I just have to make it go to the next case, so incrementing by one is as good in this situation. Speaking of incrementing by one, i should use ++ instead.

Reply:
I mean it won't make any difference to the code, but for readability you should always prefer to use defines over magic numbers